data_processing.ipynb

```diff
@@ -748,6 +820,726 @@
+      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
+     ]
+    }
+   ],
+   "source": [
+    "def generateTrainingImages(gpacData):\n",
+    "    \"\"\" Reads and processes all GPAC data from previous preprocessing and convert the data\n",
+    "    into images in greyscale and in a heatmap.\n",
+    "    \"\"\"\n",
+    "    plt.axis('off')\n",
+    "    for file in gpacData.keys():\n",
+    "        for i, gpac in enumerate(gpacData[file]['data']['fv']):\n",
+    "            label = gpacData[file]['data']['label'][i]\n",
+    "            # Window labels: baseline interictal (0), blink (1), movement (2), or EMG (3). -1\n",
+    "            if label != -1:\n",
+    "                if label == 0:\n",
+    "                    folder = 'Baseline'\n",
+    "                elif label == 1:\n",
+    "                    folder = 'Blink'\n",
+    "                elif label == 2:\n",
+    "                    folder = 'Movement'\n",
+    "                else:\n",
+    "                    folder = 'EMG'\n",
+    "                #saves all of the images in a heatmap\n",
+    "                plt.imshow(gpac)\n",
+    "                plt.savefig('data/images/' + folder  + '/' + file.split('/')[2] + str(i) + '.png', bbox_inches='tight', pad_inches=0)\n",
+    "                plt.imshow(gpac, cmap='gray')\n",
+    "                plt.savefig('data/greyscaleimages/' + folder  + '/' + file.split('/')[2] + str(i) + '.png', bbox_inches='tight', pad_inches=0)\n",
+    "\n",
+    "generateTrainingImages(process_gpac_image_data(\"data/GPAC_Images_Labelled/\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 104,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = []\n",
+    "labels = []\n",
+    "\n",
+    "from keras.preprocessing.image import img_to_array\n",
+    "from keras.preprocessing.image import load_img\n",
+    "\n",
+    "\n",
+    "\n",
+    "directory = 'data/images/Baseline'\n",
+    "for filename in os.listdir(directory):\n",
+    "    if filename.endswith(\".png\"):\n",
+    "        img = img_to_array(load_img(os.path.join(directory, filename)))\n",
+    "        data.append(img)\n",
+    "        y=0\n",
+    "        labels.append(y)\n",
+    "    else:\n",
+    "        continue\n",
```

Comment:
This code is essentially repeated 4 times, for each different directory/label. Please either make a function for it or write the loop once instead of repeating it.